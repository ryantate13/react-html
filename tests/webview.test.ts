import {webview} from '../src';

describe('webview', () => {
    it('renders with children', () => {
        expect(webview({}, 'TEST')).toMatchSnapshot();
    });
    
    it('renders arrays of children', () => {
        expect(webview({}, Array(10).fill('TEST'))).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(webview({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(webview()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(webview({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(webview({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(webview({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(webview({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(webview({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(webview({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(webview({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(webview({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(webview({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(webview({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(webview({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(webview({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(webview({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(webview({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(webview({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden=true', () => {
        expect(webview({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden=false', () => {
        expect(webview({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(webview({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(webview({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(webview({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(webview({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(webview({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(webview({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(webview({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(webview({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(webview({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(webview({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(webview({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(webview({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(webview({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope=true', () => {
        expect(webview({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope=false', () => {
        expect(webview({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(webview({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(webview({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(webview({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(webview({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(webview({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(webview({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style={"color":"#000000"}', () => {
        expect(webview({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex=1', () => {
        expect(webview({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(webview({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(webview({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(webview({translate: "no"})).toMatchSnapshot();
    });
});