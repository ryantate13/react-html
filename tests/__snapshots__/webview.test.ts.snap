// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webview renders arrays of children 1`] = `
<webview>
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
</webview>
`;

exports[`webview renders with accessKey="a,b,c,d" 1`] = `
<webview
  accessKey="a,b,c,d"
/>
`;

exports[`webview renders with autoCapitalize="characters" 1`] = `
<webview
  autoCapitalize="characters"
/>
`;

exports[`webview renders with autoCapitalize="none" 1`] = `
<webview
  autoCapitalize="none"
/>
`;

exports[`webview renders with autoCapitalize="off" 1`] = `
<webview
  autoCapitalize="off"
/>
`;

exports[`webview renders with autoCapitalize="on" 1`] = `
<webview
  autoCapitalize="on"
/>
`;

exports[`webview renders with autoCapitalize="sentences" 1`] = `
<webview
  autoCapitalize="sentences"
/>
`;

exports[`webview renders with autoCapitalize="words" 1`] = `
<webview
  autoCapitalize="words"
/>
`;

exports[`webview renders with children 1`] = `
<webview>
  TEST
</webview>
`;

exports[`webview renders with className="foo bar baz" 1`] = `
<webview
  className="foo bar baz"
/>
`;

exports[`webview renders with contentEditable="false" 1`] = `
<webview
  contentEditable="false"
/>
`;

exports[`webview renders with contentEditable="true" 1`] = `
<webview
  contentEditable="true"
/>
`;

exports[`webview renders with dir="auto" 1`] = `
<webview
  dir="auto"
/>
`;

exports[`webview renders with dir="ltr" 1`] = `
<webview
  dir="ltr"
/>
`;

exports[`webview renders with dir="rtl" 1`] = `
<webview
  dir="rtl"
/>
`;

exports[`webview renders with draggable="false" 1`] = `
<webview
  draggable="false"
/>
`;

exports[`webview renders with draggable="true" 1`] = `
<webview
  draggable="true"
/>
`;

exports[`webview renders with empty attributes 1`] = `<webview />`;

exports[`webview renders with hidden=false 1`] = `
<webview
  hidden={false}
/>
`;

exports[`webview renders with hidden=true 1`] = `
<webview
  hidden={true}
/>
`;

exports[`webview renders with id="TEST" 1`] = `
<webview
  id="TEST"
/>
`;

exports[`webview renders with inputMode="decimal" 1`] = `
<webview
  inputMode="decimal"
/>
`;

exports[`webview renders with inputMode="email" 1`] = `
<webview
  inputMode="email"
/>
`;

exports[`webview renders with inputMode="none" 1`] = `
<webview
  inputMode="none"
/>
`;

exports[`webview renders with inputMode="numeric" 1`] = `
<webview
  inputMode="numeric"
/>
`;

exports[`webview renders with inputMode="search" 1`] = `
<webview
  inputMode="search"
/>
`;

exports[`webview renders with inputMode="tel" 1`] = `
<webview
  inputMode="tel"
/>
`;

exports[`webview renders with inputMode="text" 1`] = `
<webview
  inputMode="text"
/>
`;

exports[`webview renders with inputMode="url" 1`] = `
<webview
  inputMode="url"
/>
`;

exports[`webview renders with is="foo" 1`] = `
<webview
  is="foo"
/>
`;

exports[`webview renders with itemID="https://google.com/" 1`] = `
<webview
  itemID="https://google.com/"
/>
`;

exports[`webview renders with itemProp="https://google.com/" 1`] = `
<webview
  itemProp="https://google.com/"
/>
`;

exports[`webview renders with itemRef="foo bar" 1`] = `
<webview
  itemRef="foo bar"
/>
`;

exports[`webview renders with itemScope=false 1`] = `
<webview
  itemScope={false}
/>
`;

exports[`webview renders with itemScope=true 1`] = `
<webview
  itemScope={true}
/>
`;

exports[`webview renders with itemType="https://google.com/" 1`] = `
<webview
  itemType="https://google.com/"
/>
`;

exports[`webview renders with lang="" 1`] = `
<webview
  lang=""
/>
`;

exports[`webview renders with lang="es-419" 1`] = `
<webview
  lang="es-419"
/>
`;

exports[`webview renders with slot="TEST" 1`] = `
<webview
  slot="TEST"
/>
`;

exports[`webview renders with spellCheck="false" 1`] = `
<webview
  spellCheck="false"
/>
`;

exports[`webview renders with spellCheck="true" 1`] = `
<webview
  spellCheck="true"
/>
`;

exports[`webview renders with style={"color":"#000000"} 1`] = `
<webview
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`webview renders with tabIndex=1 1`] = `
<webview
  tabIndex={1}
/>
`;

exports[`webview renders with title="TEST" 1`] = `
<webview
  title="TEST"
/>
`;

exports[`webview renders with translate="no" 1`] = `
<webview
  translate="no"
/>
`;

exports[`webview renders with translate="yes" 1`] = `
<webview
  translate="yes"
/>
`;

exports[`webview renders without attributes 1`] = `<webview />`;
